### PR TITLE
Fix bug with one-third wide columns, fixes dash-44

### PR DESCRIPTION
**Pull request by rogozinds**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-board/18)

### CODE REVIEW

File: vaadin-board-row.html [hunk 1]
```diff
@@ -55,20 +55,26 @@
        * Calculates flex basis based on colSpan and width
        * @param{Number} colspan value of the row
        * @param{Number} width of the row in px
-       * @param{Number} amount of nodes in the row
+       * @param{Number} amount of columns in the row
        */
-      _calculateFlexBasis(colSpan, width, nodesCount) {
+      _calculateFlexBasis(colSpan, width, colsInRow) {
         let flexBasis = 25 * colSpan;
```
Comment (Peppe) [inline vaadin-board-row.html]: I do not see an execution path where this calculation is actually used. Every possible path below overwrites it

**Comment by alvarezguille** [comment]
Reviewed 1 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-board-row.html, line 61 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-board/18#-KdQkQVZ3ZW6Y3ZAWN3I-r1-61:-KdQxQIUZaVleQTm0jiQ:b6t58yc) ([raw file](https://github.com/vaadin/vaadin-board/blob/58ddf51e43c7eb8b4e3ba4dae1a7ef23fce1b895/vaadin-board-row.html#L61)):*
<details><summary><i>Previously, Peppe (Jens Jansson) wrote…</i></summary><blockquote>

I do not see an execution path where this calculation is actually used. Every possible path below overwrites it
</blockquote></details>

true that

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-board/18)*

**Comment by rogozinds** [comment]
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-board-row.html, line 61 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-board/18#-KdQkQVZ3ZW6Y3ZAWN3I-r1-61:-KdQyw6mOGQRtN_cIy8A:bd7rpxp) ([raw file](https://github.com/vaadin/vaadin-board/blob/58ddf51e43c7eb8b4e3ba4dae1a7ef23fce1b895/vaadin-board-row.html#L61)):*
<details><summary><i>Previously, alvarezguille (Guille) wrote…</i></summary><blockquote>

true that
</blockquote></details>

You need to declare a variable outside the if statement so you can use it. Better to have some value than to have undefined.
I'll set it to 0.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-board/18)*

**Comment by alvarezguille** [comment]
lgtm, can you add mention to github issue too? it can be in the PR

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-board/18#-:-KdR-79eEz0HsgTIP-JD:b-5cyucv)*

File: vaadin-board-row.html [hunk 2]
```diff
@@ -55,20 +55,26 @@
        * Calculates flex basis based on colSpan and width
        * @param{Number} colspan value of the row
        * @param{Number} width of the row in px
-       * @param{Number} amount of nodes in the row
+       * @param{Number} amount of columns in the row
        */
-      _calculateFlexBasis(colSpan, width, nodesCount) {
-        let flexBasis = 25 * colSpan;
+      _calculateFlexBasis(colSpan, width, colsInRow) {
```
Comment (Peppe) [inline vaadin-board-row.html:84]: Maybe an easier way to write the whole _calculateFlexBasis:

```
_calculateFlexBasis(colSpan, width, colsInRow) {
  if (width < this._ONE_COLUMN_MAX_WIDTH) {
    colsInRow = 1;
  } else if (width < this._TWO_COLUMNS_MAX_WIDTH && colsInRow == 4) {
    colsInRow = 2;
  }
  let flexBasis = colSpan / colsInRow * 100;
  flexBasis = flexBasis > 100 ? 100 : flexBasis;
  return flexBasis + '%';
}
```
Basically - if in "small" size, always use one column, if in "medium size" and you have 4 columns, use 2 columns, otherwise calculate with the basic formula.

**Comment by rogozinds** [comment]
Review status: 1 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[vaadin-board-row.html, line 60 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-board/18#-KdQkQVZ3ZW6Y3ZAWN3I-r2-60:-KdUez8w-VKn-U3GiDSH:bb6un1j) ([raw file](https://github.com/vaadin/vaadin-board/blob/9af872d8de0592c6e8184976fe71d49faa180c7b/vaadin-board-row.html#L60)):*
<details><summary><i>Previously, Peppe (Jens Jansson) wrote…</i></summary><blockquote>

Maybe an easier way to write the whole _calculateFlexBasis:

```
_calculateFlexBasis(colSpan, width, colsInRow) {
  if (width < this._ONE_COLUMN_MAX_WIDTH) {
    colsInRow = 1;
  } else if (width < this._TWO_COLUMNS_MAX_WIDTH && colsInRow == 4) {
    colsInRow = 2;
  }
  let flexBasis = colSpan / colsInRow * 100;
  flexBasis = flexBasis > 100 ? 100 : flexBasis;
  return flexBasis + '%';
}
```
Basically - if in "small" size, always use one column, if in "medium size" and you have 4 columns, use 2 columns, otherwise calculate with the basic formula.
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-board/18)*

**Comment by rogozinds** [comment]
Fixes #4.

**Comment by alvarezguille** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-board/18)*
